agent/consul/state/catalog.go

```diff
@@ -3717,6 +3730,42 @@ func updateGatewayNamespace(tx WriteTxn, idx uint64, service *structs.GatewaySer
 	return nil
 }
 
+// serviceConnectInstances returns whether the service has at least one connect instance,
+// and at least one non-connect instance.
+func serviceConnectInstances(tx WriteTxn, serviceName string, entMeta *acl.EnterpriseMeta) (bool, bool, error) {
```

Comment:
It is a little difficult to connect the function name to what it does. So how about renaming it like `serviceHasConnectInstances`?